File: desh/ege2026kp/12solve/12-233.cpp
// Автор: М. Нехорошева

#include<iostream>
#include<string>
using namespace std;
int main(){
	string s(107,'8');
	
	for(;s.find("555")!=-1 || s.find("888")!=-1;){
		if(s.find("555")!=-1)
			s.replace(s.find("555"),3,"8");
		if(s.find("888")!=-1)
			s.replace(s.find("888"),3,"5");
	}
	cout<<s;
}

File: desh/ege2026kp/12solve/12-233.py

s = '1121121121121121111111111111'
print(s.count('1'), s.count('2'))

while '11' in s:
  if '112' in s:
    s = s.replace( '112', '5', 1 )
  else:
    s = s.replace( '11', '3' )
  print( s )

print( s )
print( sum( map(int,s) )  )
